integration_tests/src/main/python/spark_init_internal.py

```diff
@@ -154,6 +155,34 @@ def _handle_derby_dir(sb, driver_opts, wid):
         os.makedirs(d)
     sb.config('spark.driver.extraJavaOptions', driver_opts + ' -Dderby.system.home={}'.format(d))
 
+logger = logging.getLogger('__pytest_worker_logger__')
+def _configure_log_dir(_sb, wid):
+    current_directory = os.path.abspath(os.path.curdir)
+    log_file = '{}/{}_worker_logs.log'.format(current_directory, wid)
+
+    from conftest import get_std_input_path
+    std_input_path = get_std_input_path()
+    driver_opts = ' -Dlog4j.configuration=file://{}/xdist_it_log4j.properties '.format(std_input_path) + \
+        ' -Dlogfile={}'.format(log_file)
+
+    # Set up Logging
+    # Create a named logger
+    global logger
+    logger.setLevel(logging.INFO)
```

Comment:
is there a way to change log level without modifying this file?

Reply:
We could use a file config based on the [documentation](https://docs.python.org/3/library/logging.config.html#logging.config.fileConfig) but I am not sure if adding another config will add any value as this is a python file and can be changed without having to build the project.

Reply:
so its not clear to me what this setLevel is doing compared to the file_handler, compared to the log4j file?  you are passing the log4j as the driver options so why/when are these ones needed?

Reply:
Here are some points to clarify 
* The pytest and the plugin are both writing to the same log file e.g `gw0_worker_logs.log`. 
* This `logLevel` is independent of the log level set in the log4j.properties and is only used for setting the level for the `file_handler`
* The `file_handler` defined here, and thus the level, is only used for logging the test name that is needed in the worker_logs.

Reply:
so this one is just for the pytest driver/framework to write to the logs?  Above we modify the spark driver_opts so when the spark session is created it uses the log4j.properties file specified there, correct?  If that is the case just adding that as comment to this section would be nice.

What happens to logs if running in on distributed Spark cluster - like yarn or standalone mode?  I assume this will only write pytest related logs to this new file, do these configs mess up the normal spark logging?

Reply:
I answered some here. https://github.com/NVIDIA/spark-rapids/pull/10644#issuecomment-2079724449

>What happens to logs if running in on distributed Spark cluster - like yarn or standalone mode? I assume this will only write pytest related logs to this new file, do these configs mess up the normal spark logging?

I have tested this on standalone mode and the changes have no effect on the logs if a user sets the `--master` I have not tested on yarn but I assume it should be the same?

Reply:
yeah I think its fairly safe let the nightly integration tests catch any issues